artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -306,10 +306,14 @@ private boolean isPubSub(Source source) {
    @Override
    public void close(ErrorCondition condition) throws ActiveMQAMQPException {
       closed = true;
+      if (condition != null) {
+         sender.setCondition(condition);
+      }
       protonSession.removeSender(sender);
       synchronized (connection.getLock()) {
          sender.close();
       }
+      //sender.setSource(null);
```

Comment:
Why comment this out vs just removing it if it is unnecessary? Source control will retain a history of the changes